src/main/java/com/commercetools/sync/sdk2/commons/utils/CustomUpdateActionUtils.java

```diff
@@ -497,9 +499,10 @@ List<ResourceUpdateActionT> buildNewOrModifiedCustomFieldsUpdateActions(
     return newCustomFields.keySet().stream()
         .filter(
             newCustomFieldName -> {
-              final Object newCustomFieldValue = newCustomFields.get(newCustomFieldName);
-              final Object oldCustomFieldValue = oldCustomFields.get(newCustomFieldName);
-              // TODO: This comparison doesn't work when newFieldValue is passed as JSON string.
+              final Object newCustomFieldValue =
```

Comment:
```suggestion
              final JsonNode newCustomFieldValue =
```

Reply:
Also for `isNullJsonValue()` you can accept `JsonNode` instead of `Object` and remove the type checking from there.

Reply:
[a5de830](https://github.com/commercetools/commercetools-sync-java/pull/987/commits/a5de8303e2ce7084a05ce7343a1dd0baac0528ff)